async_upnp_client/server.py

```diff
@@ -1073,6 +1076,7 @@ async def _parse_action_body(
     try:
         _, action_name = soap_action.split("#")
         data = await request.text()
+        data = data.replace('"s:', '" s:').replace('"u:', '" u:')
```

Comment:
Why did you include this?

Reply:
The Amazon Alexa is sending the following data:

```
POST /upnp/control/basicevent1 HTTP/1.1
Host: 192.168.xx.xx:1900
Content-Type: text/xml; charset="utf-8"
SOAPACTION: "urn:Belkin:service:basicevent:1#GetBinaryState"
Content-Length: 298

<?xml version="1.0" encoding="utf-8"?><s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/"s:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/"><s:Body><u:GetBinaryState xmlns:u="urn:Belkin:service:basicevent:1"><BinaryState>1</BinaryState></u:GetBinaryState></s:Body></s:Envelope>
```

If you take a deep look, there is an error at `nvelope/"s:enco`.

So this code prevent the error parsing the XML.

Reply:
I'm not opposed on work-arounds, but can the work-around be applied only for this specific device? The `SOAPACTION` looks like a good candidate. I.e., something like this perhaps?
```python
service, action_name = soap_action.split("#")
...
if service == "urn:Belkin:service:basicevent:1":
  data = data.replace('"s:', '" s:')
...
```

Also, the replace for `"u:"`does not seem to be used. At least not from your example.

Finally, I'd suggest notifying Belkin about this bug. Perhaps they are willing to fix this, improving their product(s).